crates/orchestrator/src/jobs/mod.rs

```diff
@@ -343,7 +349,11 @@ pub async fn verify_job(id: Uuid, config: Arc<Config>) -> Result<(), JobError> {
         KeyValue::new("job", format!("{:?}", job)),
     ];
 
-    ORCHESTRATOR_METRICS.block_gauge.record(job.internal_id.parse::<f64>().unwrap(), &attributes);
+    let internal_id = job
+        .internal_id
+        .parse::<f64>()
+        .map_err(|e| JobError::Other(OtherError::from(format!("Could not parse internal id into f64: {e}"))))?;
```

Comment:
given this code is repeated 3 times, how about a function?

Reply:
Eventhoug it is repeated, it is just a single line. Also if I abstract in a function which returns a `color_eyre::Result` I will have to still catch and `map_err` wherever I call, in way making the intention less effective. What is your take on this?

Reply:
we could technically make `parse_string_to_u64` that returns a `JobError`. so you can call it like `parse_string_to_u64(job.internal_id)?` right?

Reply:
Yeah this could be a good solution actually, updated